core/src/main/java/org/apache/iceberg/expressions/ExpressionParser.java

```
@@ -0,0 +1,257 @@
+    generator.writeStartObject();
+    generator.writeFieldName(TYPE);
+    generator.writeStartArray();
+    for (int i = 0; i < literals.size(); i++) {
+      generator.writeString(toJsonType(literals.get(i)));
+    }
+
+    generator.writeEndArray();
+    generator.writeFieldName(VALUE);
+    generator.writeStartArray();
+    for (int i = 0; i < literals.size(); i++) {
+      generator.writeString(literals.get(i).value().toString());
+    }
+
+    generator.writeEndArray();
+    generator.writeEndObject();
+  }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  public static String toJsonType(Literal literal) throws IOException {
+    if (literal instanceof Literals.AboveMax) {
+      return "above-max";
+    } else if (literal instanceof Literals.BelowMin) {
+      return "below-min";
+    } else if (literal instanceof Literals.BooleanLiteral) {
+      return "boolean";
+    } else if (literal instanceof Literals.IntegerLiteral) {
+      return "integer";
+    } else if (literal instanceof Literals.LongLiteral) {
+      return "long";
+    } else if (literal instanceof Literals.FloatLiteral) {
+      return "float";
+    } else if (literal instanceof Literals.DoubleLiteral) {
+      return "double";
+    } else if (literal instanceof Literals.DateLiteral) {
+      return "date";
+    } else if (literal instanceof Literals.TimeLiteral) {
+      return "time";
+    } else if (literal instanceof Literals.TimestampLiteral) {
+      return "timestamp";
+    } else if (literal instanceof Literals.DecimalLiteral) {
+      return "decimal";
+    } else if (literal instanceof Literals.StringLiteral) {
+      return "string";
+    } else if (literal instanceof Literals.UUIDLiteral) {
+      return "uuid";
+    } else if (literal instanceof Literals.FixedLiteral) {
+      return "fixed";
+    } else if (literal instanceof Literals.BinaryLiteral) {
+      return "binary";
+    }
+    return "na";
```

this should throw exception, instead of returning na